config.go

```diff
@@ -50,17 +45,52 @@ type Validator interface {
 // it goes through the struct and sets up corresponding flags to be used during parsing
 func New(c interface{}) *goConfig {
 	return &goConfig{
-		envEnabled:  true,
-		fileEnabled: true,
-		tomlEnabled: true,
-		yamlEnabled: true,
-		jsonEnabled: true,
-		flagEnabled: true,
+		options: defaultOpts,
 
 		config: c,
 	}
 }
 
+type Options uint64
+
+const (
+	OptEnv Options = 1 << iota
+	OptToml
+	OptYaml
+	OptJson
+	OptFlag
+	OptGenConf // -g to generate config files
+	OptShow    // -show to show the set config values
+)
+const OptFiles = OptToml | OptYaml | OptJson
+const defaultOpts = OptEnv | OptFiles | OptFlag | OptShow | OptGenConf
+
+// Disable Options. By Default all Options are enabled.
+// OptEnv: ignore environment variables
+// OptFiles: ignore supported config files
+// OptYaml: ignore yaml config files
+// OptJson: ignore json config files
+// OptToml: ignore toml config files
+// OptFlag: ignore flag config files
+// OptGenConf: remove flag option to generate config files
+// OptShow: remove flag option to print of config values
+func (g *goConfig) Disable(opts Options) *goConfig {
+	g.options ^= opts
+	return g
+}
```

Comment:
From what I now understand of bitwise logic. This will disable a flag if it's enabled, but it will also enable it if called again on the same option.
```go
func (g *goConfig) Disable(opts Options) *goConfig {
	g.options &^= opts // This will strictly disable the specified flags
	return g
}
```
This approach ensures the flags in opts are disabled without the possibility of accidentally re-enabling them if they were already disabled.